### PR TITLE
LWDEV-3829 optimize order books

### PR DESCRIPTION
**Pull request by andrey-anokhin**

### CODE REVIEW

File: src/MarginTrading.Backend.Core/Orderbooks/OrderBookList.cs [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                return new MatchedOrderCollection();
+
+            return new MatchedOrderCollection(_orderBooks[order.Instrument]
+                .Match(order, orderTypeToMatch, volumeToMatch, _marginSettings.MaxMarketMakerLimitOrderAge));
+        }
+
+        public void Update(Order order, OrderDirection orderTypeToMatch, IEnumerable<MatchedOrder> matchedOrders)
+        {
+            if (!_orderBooks.ContainsKey(order.Instrument))
+                return;
```
Comment (GSailen) [inline src/MarginTrading.Backend.Core/Orderbooks/OrderBookList.cs]: Consider using 
if (!_orderBooks.TryGetValue(order.Instrument, out var result))
                return;

File: src/MarginTrading.Backend/Application.cs [hunk 2]
```diff
@@ -59,7 +59,7 @@ public sealed class Application
             try
             {
                 _rabbitMqService.Subscribe<MarketMakerOrderCommandsBatchMessage>(
-                    _marginSettings.MarketMakerRabbitMqSettings, _marginSettings.Env, HandleNewOrdersMessage);
+                    _marginSettings.MarketMakerRabbitMqSettings, "test", HandleNewOrdersMessage);
```
Comment (GSailen) [inline src/MarginTrading.Backend/Application.cs]: "test" ??

File: src/MarginTrading.Backend.Core/LykkeConstants.cs [hunk 3]
```diff
@@ -1,37 +1,16 @@
...
-
-        public const int TotalLykkeAmount = 1250000000;
 
-        public const int MinPwdLength = 6;
-        public const int MaxPwdLength = 100;
-
-        public const int DefaultRefundTimeoutDays = 30;
-
-        public static readonly string[] BaseAssets = { "USD", "CHF", "EUR" };
         public static decimal DefaultDemoBalance = 50000;
```
[the remark this answers is not in the document]
Comment (andrey-anokhin) [reply src/MarginTrading.Backend.Core/LykkeConstants.cs]: Probably copy-past from BaseAssets ) Fixed.

File: src/MarginTrading.Backend.Core/Orderbooks/OrderBook.cs [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+            if (volumeToMatch == 0)
+                yield break;
+
+            var source = orderTypeToMatch == OrderDirection.Buy ? Buy : Sell;
+            volumeToMatch = Math.Abs(volumeToMatch);
+            var minMarketMakerOrderDate = maxMarketMakerLimitOrderAge > 0
+                ? DateTime.UtcNow.AddSeconds(-maxMarketMakerLimitOrderAge)
+                : DateTime.MinValue;
+
+            foreach (KeyValuePair<decimal, List<LimitOrder>> pair in source)
```
[the remark this answers is not in the document]
Comment (andrey-anokhin) [reply src/MarginTrading.Backend.Core/Orderbooks/OrderBook.cs]: Used var everywhere in solution

File: src/MarginTrading.Backend.Core/Orderbooks/OrderBookList.cs (hunk 1 above)
Comment (andrey-anokhin) [reply src/MarginTrading.Backend.Core/Orderbooks/OrderBookList.cs]: Changed

File: src/MarginTrading.Backend/Application.cs (hunk 2 above)
Comment (andrey-anokhin) [reply src/MarginTrading.Backend/Application.cs]: Commited after local tests. Fixed.

**Pull request merged**